packages/flame/lib/src/widgets/animation_widget.dart

```diff
@@ -140,17 +139,19 @@ class _InternalSpriteAnimationWidgetState
   }
 
   void _setupController() {
-    _controller?.dispose();
+    widget.animation.onComplete = _pauseAnimation;
+    _controller ??= AnimationController(vsync: this)
+      ..removeListener(_onAnimationValueChanged)
+      ..addListener(_onAnimationValueChanged);
```

Comment:
I wonder, if you use the `??=` operator, will it call the `removeListener` and `addListener` only if `_controller` was null? Or will it always call it?

Because I read this code as: If `_controller` is null assign it this object and call these methods on that object

If that is the case then this won't do what you expect it to do right?

Reply:
You are right
It should be as follows.
```dart
(_controller ??= AnimationController(vsync: this))
      ..removeListener(_onAnimationValueChanged)
      ..addListener(_onAnimationValueChanged);
```

I am very confused about this after this review because even though the code has a bug but it works by either test or real build.

And I realized that removing and re-adding the listener doesn't needed.
`widget.animation` in `_onAnimationValueChanged` already referencing the updated animation after the lifecycle event.

Reply:
I removed the `removeListener` 
Thanks for the advice! :)